File: frontend/src/types/service.ts
export interface ServicePort {
  port: number;
  target_port: string | null;
  node_port: number | null;
  protocol: string;
}

export interface ServiceInfo {
  name: string;
  namespace: string;
  type: string;
  cluster_ip: string | null;
  external_ip: string | null;
  ports: ServicePort[];
  selector: Record<string, string>;
  created_at: string | null;
}

export interface IngressPath {
  path: string;
  backend_service: string | null;
  backend_port: number | null;
}

export interface IngressRule {
  host: string | null;
  paths: IngressPath[];
}

export interface IngressInfo {
  name: string;
  namespace: string;
  rules: IngressRule[];
  tls_hosts: string[];
  ingress_class: string | null;
  created_at: string | null;
}

File: backend/app/schemas/service.py
from datetime import datetime

from pydantic import BaseModel


class ServicePort(BaseModel):
    port: int
    target_port: str | None
    node_port: int | None
    protocol: str


class ServiceInfo(BaseModel):
    name: str
    namespace: str
    type: str
    cluster_ip: str | None
    external_ip: str | None
    ports: list[ServicePort]
    selector: dict[str, str]
    created_at: datetime | None


class IngressPath(BaseModel):
    path: str
    backend_service: str | None
    backend_port: int | None


class IngressRule(BaseModel):
    host: str | None
    paths: list[IngressPath]


class IngressInfo(BaseModel):
    name: str
    namespace: str
    rules: list[IngressRule]
    tls_hosts: list[str]
    ingress_class: str | None
    created_at: datetime | None
